hdf5_helper should (probably) use hsize_t instead of unsigned long long

<!--
If you are filing an issue that is not a bug report, please feel free to erase
this template and describe the issue as clearly as possible.
-->


### Summary of Problem
<!--
What behavior did you observe when encountering this issue?
What behavior did you expect to observe?
Is this a blocking issue with no known work-arounds?
-->

The HDF5 module does not compile with HDF5 version 1.14 due to mismatch in pointer types:

```
/nix/store/mj7835g4y4wspvaxfq0smljas0ibdg1a-chapel-1.31.0/modules/packages/HDF5Helper/hdf5_helper.h:55:31: error: incompatible pointer types passing 'unsigned long long *' to parameter of type 'hsize_t *' (aka 'unsigned long *') [-Werror,-Wincompatible-pointer-types]
                              (unsigned long long*)dims,
                              ^~~~~~~~~~~~~~~~~~~~~~~~~
/nix/store/k9kz55v0zv6m2klw0105agy99b5jjnmn-hdf5-1.14.0-dev/include/H5LTpublic.h:591:84: note: passing argument to parameter 'dims' here
H5_HLDLL herr_t H5LTget_dataset_info(hid_t loc_id, const char *dset_name, hsize_t *dims,
                                                                                   ^
In file included from <built-in>:5:
/nix/store/mj7835g4y4wspvaxfq0smljas0ibdg1a-chapel-1.31.0/modules/packages/HDF5Helper/hdf5_helper.h:69:27: error: incompatible pointer types passing 'unsigned long long *' to parameter of type 'const hsize_t *' (aka 'const unsigned long *') [-Werror,-Wincompatible-pointer-types]
                          (unsigned long long*)dims,
                          ^~~~~~~~~~~~~~~~~~~~~~~~~
/nix/store/k9kz55v0zv6m2klw0105agy99b5jjnmn-hdf5-1.14.0-dev/include/H5LTpublic.h:190:96: note: passing argument to parameter 'dims' here
H5_HLDLL herr_t H5LTmake_dataset(hid_t loc_id, const char *dset_name, int rank, const hsize_t *dims
```

### Suggested fix

Replace `unsigned long long` with `hsize_t`. I can make a PR.

### Steps to Reproduce

**Source Code:**

```chapel
import HDF5;
```

**Compile command:**

`chpl -Ipath/to/hdf5/include test.chpl -Lpath/to/hdf5/lib -lhdf5_hl -lhdf5` 


### Configuration Information

- Output of `chpl --version`: chpl version 1.31.0 pre-release (xxxxxxxxxx)
- Output of `$CHPL_HOME/util/printchplenv --anonymize`:
```
CHPL_TARGET_PLATFORM: linux64
CHPL_TARGET_COMPILER: llvm
CHPL_TARGET_ARCH: x86_64
CHPL_TARGET_CPU: none *
CHPL_LOCALE_MODEL: flat
CHPL_COMM: none
CHPL_TASKS: qthreads
CHPL_LAUNCHER: none
CHPL_TIMERS: generic
CHPL_UNWIND: system *
CHPL_MEM: jemalloc
CHPL_ATOMICS: cstdlib
CHPL_GMP: system *
CHPL_HWLOC: bundled
CHPL_RE2: bundled *
CHPL_LLVM: system *
CHPL_AUX_FILESYS: none
```
- Back-end compiler and version, e.g. `clang --version`: 14

--- a/modules/packages/HDF5Helper/hdf5_helper.h
+++ b/modules/packages/HDF5Helper/hdf5_helper.h
@@ -52,7 +52,7 @@ static herr_t H5LTget_dataset_info_WAR(hid_t loc_id,
                                        size_t* type_size) {
   return H5LTget_dataset_info(loc_id,
                               dset_name,
-                              (unsigned long long*)dims,
+                              (hsize_t*)dims,
                               type_class,
                               type_size);
 }
@@ -66,7 +66,7 @@ static herr_t H5LTmake_dataset_WAR(hid_t loc_id,
   return H5LTmake_dataset(loc_id,
                           dset_name,
                           rank,
-                          (unsigned long long*)dims,
+                          (hsize_t*)dims,
                           type_id,
                           buffer);
 }

